### PR TITLE
Update step certificates and cli

### PR DESCRIPTION
**Pull request by dguibert**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124347` at c2028794 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>step-ca</li>
    <li>step-cli</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/tools/security/step-cli/default.nix:4:3:
```
  |
4 | , fetchpatch
  |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/tools/security/step-cli/default.nix [hunk 1]
```diff
@@ -6,21 +6,21 @@
 
 buildGoModule rec {
   pname = "step-cli";
-  version = "0.15.3-22-g3ddc5aa";
+  version = "0.15.16";
 
   # 0.15.3 isn't enough, because we need https://github.com/smallstep/cli/pull/394
```
Comment (xfix) [inline pkgs/tools/security/step-cli/default.nix:11]: This comment could be removed.

**Review by sugar700 — CHANGES_REQUESTED** [review]

**Review by sugar700 — APPROVED** [review]

**Review by mohe2015 — CHANGES_REQUESTED** [review]
What about the following?
```diff
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index d186e8e27f0..a7d1c03cac7 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -3125,7 +3125,6 @@ in
 
   step-ca = callPackage ../tools/security/step-ca {
     inherit (darwin.apple_sdk.frameworks) PCSC;
-    buildGoModule = buildGo115Module;
   };
 
   step-cli = callPackage ../tools/security/step-cli { };
```

Edit: Other than that did some minimal testing with:
```
step certificate create foo foo.crt foo.key --profile root-ca
step ca init
step-ca ~/.step/config/ca.json
```

**Comment by mohe2015** [comment]
https://github.com/smallstep/certificates/blob/master/.github/workflows/release.yml They test with 1.16 also and I assume updating to the latest version is generally a good thing. @SuperSandro2000 why do you not like this suggestion?

**Comment by dguibert** [comment]
step-cli is using buildGoModule. step-ca is tested with Go 1.16 so it could be compiled by buildGoModule as well.
If both are installed on the same machine, only one go derivation will be needed.

**Comment by mohe2015** [comment]
@dguibert you commit message that updates step-cli mentions the wrong patch version. (.15 instead of .16). Otherwise LGTM

**Comment by dguibert** [comment]
> @dguibert you commit message that updates step-cli mentions the wrong patch version. (.15 instead of .16). Otherwise LGTM

Good noticed. Changed.

**Review by mohe2015 — APPROVED** [review]
LGTM!

**Review by mohe2015 — APPROVED** [review]
Still LGTM! (Stupid ofborg)

**Comment by mohe2015** [comment]
I think this should get backported to both release branches as I don't know if the updates fixed security vulerabilities or important bugs but as this is a security critical package I think we should assume so. So please add the backport labels before merging.

**Comment by SuperSandro2000** [comment]
> Still LGTM! (Stupid ofborg)

You don't need to re review it.

**Pull request merged**